ratatui/src/widgets/widget_ref.rs

```diff
@@ -82,12 +82,23 @@ pub trait WidgetRef {
     /// Draws the current state of the widget in the given buffer. That is the only method required
     /// to implement a custom widget.
     fn render_ref(&self, area: Rect, buf: &mut Buffer);
+
+    /// Converts the widget into a boxed widget.
+    fn into_boxed(self) -> Box<dyn WidgetRef>
+    where
+        Self: Sized + 'static,
+    {
+        Box::new(self)
+    }
```

Comment:
I'll remove this for now as it didn't end up being necessary - I'm unsure whether this should be on Widget or WidgetRef right now.